Packs/GoogleApigee/Integrations/GoogleApigeeEventCollector/GoogleApigeeEventCollector.py

```diff
@@ -0,0 +1,310 @@
+    :param org_name (str): the organization name
+    :param zone (str): the zone name
+    :param limit (int): the maximum logs to return per fetch call
+    """
+
+    def __init__(self, base_url: str, username: str, password: str, verify: bool, proxy: bool,
+                 org_name: str, zone: str, limit: int = DEFAULT_LIMIT, **kwargs):
+        self.username = username
+        self.password = password
+        self.org_name = org_name
+        self.max_fetch = limit
+        self.zone = zone
+
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, **kwargs)
+
+    def http_request(self, *args, **kwargs):
+        """
+        Overrides Base client request function, retrieves and adds to headers access token before sending the request.
+        """
+        token = self.get_access_token()
+        headers = {
+            'Authorization': f'Bearer {token}',
+            'Content-Type': 'application/json',
+        }
+        return super()._http_request(*args, headers=headers, **kwargs)  # type: ignore[misc]
+
+    def get_access_token(self) -> str:
+        """
+       Obtains access and refresh token from server.
+       Access token is used and stored in the integration context until expiration time.
+       After expiration, new access token are obtained and stored in the integration context.
+
+        Returns:
+            str: the access token.
+       """
+        integration_context = get_integration_context()
+        access_token = integration_context.get('access_token')
+        token_initiate_time = integration_context.get('token_initiate_time')
+        token_expiration_seconds = integration_context.get('token_expiration_seconds')
+        refresh_token = integration_context.get('refresh_token', '')
+
+        if access_token and Client.is_token_valid(
+            token_initiate_time=float(token_initiate_time),
+            token_expiration_seconds=float(token_expiration_seconds)
+        ):
+            return access_token
+        # There's no token or it is expired
+        access_token, token_expiration_seconds, refresh_token = self.get_token_request(refresh_token)
+        integration_context = {
+            'access_token': access_token,
+            'token_expiration_seconds': token_expiration_seconds,
+            'token_initiate_time': time.time(),
+            'refresh_token': refresh_token,
+        }
+        demisto.info('successfully updated access token')
+        set_integration_context(context=integration_context)
+
+        return access_token
+
+    def get_token_request(self, refresh_token: str = '') -> tuple[str, str, str]:
+        """
+        Sends request to retrieve token.
+
+       Returns:
+           tuple[str, str]: token and its expiration date
+        """
+        grant_type = 'refresh_token' if refresh_token else 'password'
+        data = {
+            'username': self.username,
+            'password': self.password,
```

Comment:
We need to send the `username` and `password` even when sending the `refresh_token`?